Error disabling authentication regulation

From the template:

```yaml
# Configuration of the authentication regulation mechanism.
#
# This mechanism prevents attackers from brute forcing the first factor.
# It bans the user if too many attempts are done in a short period of
# time.
regulation:
  # The number of failed login attempts before user is banned.
  # Set it to 0 to disable regulation.
  max_retries: 3
```

However, changing `max_retries` to 0 results in an error:
```
panic: runtime error: index out of range [0] with length 0
goroutine 35 [running]:
github.com/authelia/authelia/internal/regulation.(*Regulator).Regulate(0x400026ef40, 0x400036c338, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0)
        github.com/authelia/authelia@/internal/regulation/regulator.go:75 +0x3cc
github.com/authelia/authelia/internal/handlers.FirstFactorPost(0x400039e000)
        github.com/authelia/authelia@/internal/handlers/handler_firstfactor.go:25 +0x90
github.com/authelia/authelia/internal/middlewares.AutheliaMiddleware.func1.1(0x400037e000)
        github.com/authelia/authelia@/internal/middlewares/authelia_context.go:46 +0x114
github.com/fasthttp/router.(*Router).Handler(0x40003083c0, 0x400037e000)
        github.com/fasthttp/router@v0.5.2/router.go:326 +0xc48
github.com/authelia/authelia/internal/middlewares.LogRequestMiddleware.func1(0x400037e000)
        github.com/authelia/authelia@/internal/middlewares/log_request.go:14 +0xd4
github.com/valyala/fasthttp.(*Server).serveConn(0x4000328200, 0xf04380, 0x40002dc018, 0x0, 0x0)
        github.com/valyala/fasthttp@v1.6.0/server.go:2018 +0x4f8
github.com/valyala/fasthttp.(*workerPool).workerFunc(0x4000270aa0, 0x40002de060)
        github.com/valyala/fasthttp@v1.6.0/workerpool.go:223 +0x7c
github.com/valyala/fasthttp.(*workerPool).getCh.func1(0x4000270aa0, 0x40002de060, 0xbc3f60, 0x40002de060)
        github.com/valyala/fasthttp@v1.6.0/workerpool.go:195 +0x30
created by github.com/valyala/fasthttp.(*workerPool).getCh
        github.com/valyala/fasthttp@v1.6.0/workerpool.go:194 +0x11c
```

--- a/internal/regulation/regulator.go
+++ b/internal/regulation/regulator.go
@@ -18,7 +18,8 @@ func NewRegulator(configuration *schema.RegulationConfiguration, provider storag
 		if configuration.FindTime > configuration.BanTime {
 			panic(fmt.Errorf("find_time cannot be greater than ban_time"))
 		}
-		regulator.enabled = true
+		// Set regulator enabled only if MaxRetries is not 0.
+		regulator.enabled = configuration.MaxRetries > 0
 		regulator.maxRetries = configuration.MaxRetries
 		regulator.findTime = time.Duration(configuration.FindTime) * time.Second
 		regulator.banTime = time.Duration(configuration.BanTime) * time.Second

--- a/internal/regulation/regulator_test.go
+++ b/internal/regulation/regulator_test.go
@@ -40,7 +40,7 @@ func (s *RegulatorSuite) TearDownTest() {
 }
 
 func (s *RegulatorSuite) TestShouldNotThrowWhenUserIsLegitimate() {
-	attempts := []models.AuthenticationAttempt{
+	attemptsInDB := []models.AuthenticationAttempt{
 		models.AuthenticationAttempt{
 			Username:   "john",
 			Successful: true,
@@ -50,7 +50,7 @@ func (s *RegulatorSuite) TestShouldNotThrowWhenUserIsLegitimate() {
 
 	s.storageMock.EXPECT().
 		LoadLatestAuthenticationLogs(gomock.Eq("john"), gomock.Any()).
-		Return(attempts, nil)
+		Return(attemptsInDB, nil)
 
 	regulator := regulation.NewRegulator(&s.configuration, s.storageMock, &s.clock)
 
@@ -253,4 +253,51 @@ func (s *RegulatorSuite) TestShouldCheckRegulationHasBeenResetOnSuccessfulAttemp
 func TestRunRegulatorSuite(t *testing.T) {
 	s := new(RegulatorSuite)
 	suite.Run(t, s)
+}
+
+// This test checks that the regulator is disabled when configuration is set to 0.
+func (s *RegulatorSuite) TestShouldHaveRegulatorDisabled() {
+	attemptsInDB := []models.AuthenticationAttempt{
+		models.AuthenticationAttempt{
+			Username:   "john",
+			Successful: false,
+			Time:       s.clock.Now().Add(-31 * time.Second),
+		},
+		models.AuthenticationAttempt{
+			Username:   "john",
+			Successful: false,
+			Time:       s.clock.Now().Add(-34 * time.Second),
+		},
+		models.AuthenticationAttempt{
+			Username:   "john",
+			Successful: false,
+			Time:       s.clock.Now().Add(-36 * time.Second),
+		},
+	}
+
+	s.storageMock.EXPECT().
+		LoadLatestAuthenticationLogs(gomock.Eq("john"), gomock.Any()).
+		Return(attemptsInDB, nil)
+
+	// Check Disabled Functionality
+	configuration := schema.RegulationConfiguration{
+		MaxRetries: 0,
+		FindTime:   180,
+		BanTime:    180,
+	}
+
+	regulator := regulation.NewRegulator(&configuration, s.storageMock, &s.clock)
+	_, err := regulator.Regulate("john")
+	assert.NoError(s.T(), err)
+
+	// Check Enabled Functionality
+	configuration = schema.RegulationConfiguration{
+		MaxRetries: 1,
+		FindTime:   180,
+		BanTime:    180,
+	}
+
+	regulator = regulation.NewRegulator(&configuration, s.storageMock, &s.clock)
+	_, err = regulator.Regulate("john")
+	assert.Equal(s.T(), regulation.ErrUserIsBanned, err)
 }

